monix-tail/shared/src/test/scala/monix/tail/ArbitraryInstances.scala

```diff
@@ -96,18 +96,16 @@ trait ArbitraryInstances extends monix.eval.ArbitraryInstances {
       }
     }
 
-  implicit def isEqIterantTask[A](implicit A: Eq[A]): Eq[Iterant[Task, A]] =
+  implicit def isEqIterantTask[A](implicit A: Eq[A], sc: TestScheduler): Eq[Iterant[Task, A]] =
     new Eq[Iterant[Task, A]] {
       def eqv(lh: Iterant[Task,  A], rh: Iterant[Task,  A]): Boolean = {
-        implicit val s = TestScheduler()
```

Comment:
I had to remove these local `TestScheduler`s and require a test suite to provide one.

The reason was that, say I wanted to use `Iterant.toReactivePublisher`, which takes a `Scheduler`, and I have one in test suite. To actually run it, I need to call a `.tick()` on a test scheduler. In other methods, `equalityFuture` will internally do it, but here it calls `.tick()` on a different scheduler, and the `Future` I got from `toReactivePublisher` would not be completed.

In particular, this intuitive test:
```scala
Iterant[IO].fromReactivePublisher(it.toReactivePublisher) <-> it
```
would fail, because LHS would not be completed.